boards/xtensa/intel_adsp_cavs25/intel_adsp_cavs25.yaml

```diff
@@ -1,6 +1,7 @@
 identifier: intel_adsp_cavs25
 name: cAVS 2.5 Audio DSP (converged Audio Voice and Speech)
 type: mcu
+simulation: xt-sim
```

Comment:
and we are not a simulator.

Reply:
I try to add a type "adsp" instead of "mcu" for intel cavs board. By setting this to make twister run cavs platform by BinaryHandler.

Reply:
why? we do not want to end up with everyone defining their own type. This is not the intended use for this key. I think you are taking the wrong approach completely and fixing an issue in the wrong place. Please start a discussion first about what you are trying to do and what problem are you trying to solve.

Reply:
Thanks for the review, @nashif , let me have a simple explanation please.
Actually I want to run the CAVS testing in a by the BianryHandler, because in the DeviceHander the output by the script cannot be handled well in serial-pty mode. The BianryHandler can handle it well directly. (Base on our current script to get the cavs output)

Except adding a new type, I think there are some way to do this:
(1) Add a extra parameter to twister, which will run the specified script by BinaryHandler. (ex --binary-runner to specify the cavstwist.sh and its arguments)
(2) Add another mode in the DeviceHandler besides serial and serial-pty mode, which behaves like the BinaryHandler. This willf also be benifit if we want to add a hardware-map for it.
(3) Change some behavior of cavstwister.sh and cavstool.py to fit the serial-pty, or even add a new runner for it (Like the intel_s1000_crb did)

So I think the critical problem for me is should we only run the CAVS boards by DeviceHandler only in twister? Thanks.

Reply:
I correct the wrong usage of it in the first version. The second version still use BinaryHandler to run. And this is the third version, it runs by the DeviceHandler and support without run with -j 1 well. Also the hardware map file can be used to specify the remote-host.  But one point is I make some extra change in device-serial-pty, to make it judge the output from the stdout. Could you please help to give some feedback?